sdk/typescript/src/types/sui-bcs.ts

```diff
@@ -433,6 +449,16 @@ const BCS_0_23_SPEC = {
   },
```

Comment:
Here You should add 
```
SharedObjectRef: {
    struct: {
      objectId: 'address',
      initialSharedVersion: 'u64',
    },
  },
```
to `BCS_0_23_SPEC` since it's still using the old definition

Reply:
Alternatively, you can just change line 440 to `BCS_0_24_SPEC`

Reply:
Just moved the ref to BCS_0_23_SPEC

Reply:
I think it should be the reverse. BCS_24 should have the `SharedObjectRef` overridden while BCS_23 should have both the `SharedObjectRef` and `MoveCallTx` overridden. 

In you current implementation, both BCS_24 and BCS_23 are the same